File: VB/MyXtraGrid/DataHelper.vb
Imports System
Imports System.Collections.Generic
Imports System.Data
Imports System.Linq
Imports System.Text

Namespace MyXtraGrid
	Public Module DataHelper
		Public Function CreateTable(ByVal RowCount As Integer) As DataTable
			Dim tbl As New DataTable()
			tbl.Columns.Add("Name", GetType(String))
			tbl.Columns.Add("Rating", GetType(Integer))
			tbl.Columns.Add("Number", GetType(Integer))
			tbl.Columns.Add("Date", GetType(Date))
			Dim j As Integer = 0
			For i As Integer = 0 To RowCount - 1
				tbl.Rows.Add(New Object() { String.Format("Name{0}", i Mod 3), j, 3 - i, Date.Now.AddDays(i) })
				Dim tempVar As Boolean = j > 5
				j += 1
				If tempVar Then
					j = 0
				End If
			Next i
			Return tbl
		End Function
	End Module
End Namespace

File: CS/MyXtraGrid/Datahelper.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Text;

namespace MyXtraGrid
{
    public static class DataHelper
    {
        public static DataTable CreateTable(int RowCount)
        {
            DataTable tbl = new DataTable();
            tbl.Columns.Add("Name", typeof(string));
            tbl.Columns.Add("Rating", typeof(int));
            tbl.Columns.Add("Number", typeof(int));
            tbl.Columns.Add("Date", typeof(DateTime));
            int j = 0;
            for (int i = 0; i < RowCount; i++)
            {
                tbl.Rows.Add(new object[] { String.Format("Name{0}", i % 3), j, 3 - i, DateTime.Now.AddDays(i) });
                if (j++ > 5)
                    j = 0;
            }
            return tbl;
        }
    }
}
